nuxeo-management-rest-api/src/main/java/org/nuxeo/rest/management/probes/ProbeInfoWriter.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * (C) Copyright 2019 Nuxeo (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     Nour Al Kotob
+ */
+
+package org.nuxeo.rest.management.probes;
+
+import static org.nuxeo.ecm.core.io.registry.reflect.Instantiations.SINGLETON;
+import static org.nuxeo.ecm.core.io.registry.reflect.Priorities.REFERENCE;
+
+import java.io.IOException;
+
+import org.nuxeo.ecm.core.io.marshallers.json.AbstractJsonWriter;
+import org.nuxeo.ecm.core.io.registry.reflect.Setup;
+import org.nuxeo.ecm.core.management.api.ProbeInfo;
+
+import com.fasterxml.jackson.core.JsonGenerator;
+
+/**
+ * see {@link AbstractJsonWriter}
+ *
+ * @since 11.1
+ */
+@Setup(mode = SINGLETON, priority = REFERENCE)
+public class ProbeInfoWriter extends AbstractJsonWriter<ProbeInfo> {
+
+    @Override
+    public void write(ProbeInfo entity, JsonGenerator jg) throws IOException {
+        jg.useDefaultPrettyPrinter();
+        jg.writeStartObject();
+
+        jg.writeStringField("shortcutName", entity.getShortcutName());
+        jg.writeBooleanField("success",entity.getStatus().isSuccess());
+        jg.writeStringField("result", entity.getStatus().toString());
```

Comment:
You can't leverage `toString` in a marshaller. Purpose of this class is to have a uniq place where to put the format, using  `toString` method will externalise the behavior.

Furthermore, looking at the `toString` code, the output is not the same depending on success field and not a valid/useful json format.

See code below:
```
        if (infos == null || infos.isEmpty()) {
            return Boolean.toString(success);
        }
        if (infos.size() == 1) {
            return infos.values().iterator().next();
        }

        StringBuilder sb = new StringBuilder();
        for (String key : infos.keySet()) {
            sb.append(key);
            sb.append(" : ");
            sb.append(infos.get(key));
            sb.append("\n");
        }
        return sb.toString();
```

I think you need to write a writer for `ProbeStatus` object and call it there. You can use `AbstractJsonWriter#writeEntity` or `AbstractJsonWriter#writeEntityField` to call new writer.